brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -209,6 +212,28 @@ public void finishSpan(long duration) {
 
         internalFinishSpan(span, duration);
     }
+    
+    public <T> T trace(String component, String operation, Callable<T> callable) throws RuntimeException {
+        try {
+            startNewSpan(component, operation);
+            return callable.call();
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
The problem is that Callable.call throws Exception. If I propagate this I force people to catch it. I can improve this though to not wrap RuntimeException.